Source/RK3/RK3.H

```diff
@@ -66,6 +72,99 @@ void CalcDiffFlux(const MultiFab& cons_in,
+                           const int &spatial_order);
+
+AMREX_GPU_DEVICE
+Real InterpolateDensityFromCellToFace(const int &i, const int &j, const int &k,
+                               const Array4<Real>& cons_in,
+                               const enum NextOrPrev &nextOrPrev,
+                               const enum Coord& coordDir,
+                               const int &spatial_order);
+
+AMREX_GPU_DEVICE
+Real InterpolateRhoThetaFromCellToFace(const int &i, const int &j, const int &k,
+                                      const Array4<Real>& cons_in,
+                                      const enum NextOrPrev &nextOrPrev,
+                                      const enum Coord& coordDir,
+                                      const int &spatial_order);
+
+AMREX_GPU_DEVICE
+Real InterpolateScalarFromCellToFace(const int &i, const int &j, const int &k,
+                                      const Array4<Real>& cons_in,
+                                      const enum NextOrPrev &nextOrPrev,
+                                      const enum Coord& coordDir,
+                                      const int &spatial_order);
+
+// Meant for {x, y, z}- momentum equations
+AMREX_GPU_DEVICE
+Real ComputeAdvectedQuantityForMom(const int &i, const int &j, const int &k,
+                             const Array4<Real>& rho_u, const Array4<Real>& rho_v, const Array4<Real>& rho_w,
+                             const Array4<Real>& u, const Array4<Real>& v, const Array4<Real>& w,
+                             const enum NextOrPrev &nextOrPrev,
+                             const enum AdvectedQuantity &advectedQuantity,
+                             const enum AdvectingQuantity &advectingQuantity,
+                             const int &spatial_order);
+
+// Meant for the continuity, energy, and scalar equations
+AMREX_GPU_DEVICE
+Real ComputeAdvectedQuantityForState(const int &i, const int &j, const int &k,
+                             const Array4<Real>& rho_u, const Array4<Real>& rho_v, const Array4<Real>& rho_w,
+                             const Array4<Real>& cons_in,
+                             const enum NextOrPrev &nextOrPrev,
+                             const enum AdvectedQuantity &advectedQuantity,
+                             const enum AdvectingQuantity &advectingQuantity,
+                             const int &spatial_order);
+
+// Compute Sij
+AMREX_GPU_DEVICE
+Real ComputeStrainRate(const int &i, const int &j, const int &k,
+                        const Array4<Real>& u, const Array4<Real>& v, const Array4<Real>& w,
+                        const enum NextOrPrev &nextOrPrev,
+                        const enum MomentumEqn &momentumEqn,
+                        const enum DiffusionDir &diffDir,
+                        const Geometry &geom);
+
+// Compute turbVisc, i.e., 'nut' or Kij at cell centres. Initialize it to 1.0
+//TODO: Check if it is better to  fill all Kij and use it or compute on the fly
```

Comment:
Under a "flops are free" paradigm, I would say just compute on the fly. Something like S_ij might make more sense to precompute and store.